### PR TITLE
shipit_uplift: Switch to using async IO for coverage endpoints

### LINKED ISSUE #856: shipit_uplift: Retrieve coverage and annotate in parallel
Right now, we're retrieving annotate first and then coverage.

### LINKED ISSUE #862: shipit_uplift: Use aiohttp to perform requests
Related to #856. Instead of using multiple threads, it would be better to use async io.

### PR DESCRIPTION
**Pull request by marco-c**
Fixes #856 and fixes #862.

We are blocked on updating to aiohttp 3 because of the Taskcluster library (https://github.com/taskcluster/taskcluster-client.py/issues/96). aiohttp 3 would be needed in order to do `async with aiohttp.request` without leaks.
If we can't upgrade to aiohttp 3 soon enough, we should switch to using a aiohttp session directly (#863).

### CODE REVIEW

**Comment by garbas** [comment]
this was fixed in taskcluster client -> https://github.com/taskcluster/taskcluster-client.py/pull/100
but we would have to update taskcluster client before moving on with this PR, #685

**Comment by marco-c** [comment]
@garbas do you know of any alternative to `responses` for `aiohttp`?

**Comment by garbas** [comment]
@marco-c aiohttp comes with testclient and testserver which you can use to mock stuff. quick search online pointed me to https://github.com/pnuckowski/aioresponses but i never used it before.

**Comment by marco-c** [comment]
> @marco-c aiohttp comes with testclient and testserver which you can use to mock stuff.

I think they are only useful when you are writing aiohttp-based servers, so not in this case.

> quick search online pointed me to https://github.com/pnuckowski/aioresponses but i never used it before.

Building the derivation for `aioresponses` fails because the setup.cfg contains a non-ascii character (`author = Paweł Nuckowski`). After removing that, it fails because setup.py contains `setup_requires=['pbr'],` (`Could not find suitable distribution for Requirement.parse('pbr')`). If I remove that, I can't import aioresponses.

**Comment by marco-c** [comment]
There's also another alternative, https://github.com/CircleUp/aresponses, but it only supports Python 3.6+.

**Comment by garbas** [comment]
@marco-c it is anyway on the todo list to migrate to python 3.6, maybe shipit_uplift could be the first project? The only problem i could see is that we would have to keep `cli_common` and `backend_common` compatible with 3.5 and 3.6

**Comment by garbas** [comment]
@marco-c also I can look into getting `aioresponse` to work.

**Comment by garbas** [comment]
i got `aioresponse` to work in case you want to go this road.

**Comment by marco-c** [comment]
This should now be ready for review, but I'd like to merge it after the next release so that we have more time for testing.

File: src/shipit_uplift/shipit_uplift/coverage.py [hunk 1]
```diff
@@ -33,142 +34,112 @@ class CoverageException(Exception):
...
 
-        return r.json()
+            return await r.json()
 
     @staticmethod
-    def get_latest_build():
-        r = requests.get(CoverallsCoverage.URL + '/github/marco-c/gecko-dev.json?page=1')
-        builds = r.json()['builds']
+    async def get_latest_build():
+        async with CoverallsCoverage._get('/github/marco-c/gecko-dev.json?page=1') as r:
```
Comment (La0) [inline src/shipit_uplift/shipit_uplift/coverage.py]: I like this syntax.

**Pull request merged**